src/neo/SmartContract/ApplicationEngine.Runtime.cs

```diff
@@ -146,15 +147,16 @@ internal void RuntimeLog(byte[] state)
             Log?.Invoke(this, new LogEventArgs(ScriptContainer, CurrentScriptHash, message));
         }
 
-        internal void RuntimeNotify(StackItem state)
+        internal void RuntimeNotify(byte[] eventName, Array state)
         {
+            if (eventName.Length > MaxEventName) throw new ArgumentException();
             if (!CheckItemForNotification(state)) throw new ArgumentException();
-            SendNotification(CurrentScriptHash, state);
+            SendNotification(CurrentScriptHash, Encoding.ASCII.GetString(eventName), state);
```

Comment:
UTF8?

Reply:
Allow UTF8 in event name? I don't know...

Reply:
Me neither, but if someone want to use a Chinese characters, it won't be able to do that.

Reply:
```suggestion
            SendNotification(CurrentScriptHash, Encoding.UTF8.GetString(eventName), state);
```

Reply:
Modified.